### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by ACEof**
ICE-2 Write tests for lexer.

### CODE REVIEW

**Comment by ACEof** [comment]
Sorry. I accidentally deleted branch in my repository

File: package.json [hunk 1]
```diff
@@ -21,12 +21,14 @@
...
     "babel-preset-env": "^1.6.1",
     "babel-preset-react": "^6.24.1",
+    "eslint": "^4.19.1",
     "eslint-config-airbnb": "^16.1.0",
     "eslint-plugin-import": "^2.11.0",
     "eslint-plugin-jsx-a11y": "^6.0.3",
     "eslint-plugin-react": "^7.7.0",
     "html-webpack-plugin": "^3.1.0",
     "jest": "^22.4.3",
+    "mini-css-extract-plugin": "^0.4.0",
```
Comment (abtv) [inline package.json]: Why have you added it here?

File: src/index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import App from './components/App';
+import React from 'react'
+import ReactDOM from 'react-dom'
+import App from './components/App'
 
-ReactDOM.render(<App />, document.getElementById('app'));
+
+ReactDOM.render(<App />, document.getElementById('app'))
```
Comment (abtv) [inline src/index.js]: Please setup linter to ignore multiple lines

File: tests/lexers.test.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
+import { parseTockens } from '../src/componets/lexer'
+test ('Checking the search function for finding tokens: let a, let b ', () => {
+  expect(parseTockens('let a \
+                        a = 5 \
+                        let b')).toEqual({
+                          [
+                            {type: 'varDeclaration', value: 'let'},
```
Comment (abtv) [inline tests/lexers.test.js]: Could you create enumeration for type?

File: tests/lexers.test.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                            {type: 'identifer', value: 'a'},
+                            {type: 'identifer', value: 'a'},
+                            {type: 'varDeclaration', value: 'let'},
+                            {type: 'eqully', value: '='},
```
Comment (abtv) [inline tests/lexers.test.js]: typo: equality

File: tests/lexers.test.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                            {type: 'number', value: '5'}
+                          ]
+                        })
+})
+
+test ('Checking the search function for finding tokens: a + b', () => {
+  expect(parseTockens('a = 5 \
```
Comment (abtv) [inline tests/lexers.test.js]: typo

File: tests/lexers.test.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
+import { parseTockens } from '../src/componets/lexer'
+test ('Checking the search function for finding tokens: let a, let b ', () => {
+  expect(parseTockens('let a \
```
Comment (abtv) [inline tests/lexers.test.js]: typo

File: webpack.config.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
-const path = require('path');
-const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const MiniCssExtractPlugin = require('mini-css-extract-plugin')
```
Comment (abtv) [inline webpack.config.js]: What it is for? AFAIK we use css in js

File: webpack.config.js [hunk 8]
```diff
@@ -14,12 +15,21 @@ module.exports = {
         exclude: /node_modules/,
         use: { loader: 'babel-loader' },
         resolve: { extensions: ['*', '.js', '.jsx'] },
+      },
+      {
+        test: /\.(css)$/,
+        use: [MiniCssExtractPlugin.loader, 'css-loader']
+
```
Comment (abtv) [inline webpack.config.js]: Also setup linter to report empty lines in json (if it's possible). At least you can review your PR first

File: tests/lexers.test.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+                            {type: 'eqully', value: '='},
+                            {type: 'number', value: '5'}
+                          ]
+                        })
+})
+
+test ('Checking the search function for finding tokens: a + b', () => {
+  expect(parseTockens('a = 5 \
+                       b = 5 \
+                        a + b')).toEqual({
```
Comment (abtv) [inline tests/lexers.test.js]: What does {[...]} means here?

File: tests/lexers.test.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+})
+
+test ('Checking the search function for finding tokens: let a, a = 1.2345', () => {
+  expect(parseTockens('let a \
+                        a = 1.2345')).toEqual({
+                          [
+                            {type: 'varDeclaration', value: 'let'},
+                            {type: 'identifer', value: 'a'},
+                            {type: 'equally', value: '='},
+                            {type: 'number', value: '1.2345'}
```
Comment (abtv) [inline tests/lexers.test.js]: All these tests are for valid source code. I would like to see approach which tests incorrect source code. You need to discuss it with @AdmiralJMP1 . You need to agree what will be returned when code contains valid and invalid data. For instance, 1+2 is invalid, it should be 1 + 2 (separated by spaces). Please think how you will implement and test the requirement.

File: webpack.config.js (hunk 7 above)
Comment (ACEof) [reply webpack.config.js]: It's necessary for the correct assembly of the project by webpack.

File: src/index.js (hunk 2 above)
Comment (ACEof) [reply src/index.js]: I haven't found the requried property in the documentation ESlint.

File: webpack.config.js (hunk 8 above)
Comment (ACEof) [reply webpack.config.js]: In my opinion it's impossible.

File: tests/lexers.test.js (hunk 6 above)
Comment (abtv) [reply tests/lexers.test.js]: still not fixed

File: src/index.js (hunk 2 above)
Comment (abtv) [reply src/index.js]: In this case I think you should make it manually.

File: tests/example.test.js [hunk 11]
```diff
@@ -1,5 +1,5 @@
-import {square} from '../example'
+import { square } from '../example'
 
-test('square 5 to equal 25', () =>{
-  expect(square(5)).toBe(25);
-});
+test('square 5 to equal 25', () => {
```
Comment (abtv) [inline tests/example.test.js]: please remove this file, we don't need it

File: tests/lexers.test.js [hunk 12]
```diff
@@ -0,0 +1,146 @@
+const typeTokens = {
```
Comment (abtv) [inline tests/lexers.test.js]: please use tokenTypes

File: tests/lexers.test.js [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+  varDeclaration: 'varDeclaration',
+  dentifer: 'identifer',
```
Comment (abtv) [inline tests/lexers.test.js]: I think you need to double check your PRs.

File: webpack.config.js [hunk 14]
```diff
@@ -1,5 +1,5 @@
-const path = require('path');
-const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
```
Comment (abtv) [inline webpack.config.js]: Why didn't you remove it?

File: webpack.config.js [hunk 15]
```diff
@@ -14,12 +14,16 @@ module.exports = {
...
         resolve: { extensions: ['*', '.js', '.jsx'] },
-      }
+      },
     ]
   },
   plugins: [
     new HtmlWebpackPlugin({
       template: './src/index.html'
+    }),
+    new MiniCssExtractPlugin({
```
Comment (abtv) [inline webpack.config.js]: The same question

**Comment by ACEof** [comment]
The same tests had @AdmiralJMP1 in PR.

**Pull request closed without merging**